pgrouting 3.6.0 fails to build on OSX

conda-forge's build of pgrouting 3.6.0 failed on [Mac OSX](https://github.com/conda-forge/pgrouting-feedstock/pull/16).  The reported error appears to be

```
[121/197] Building CXX object src/cpp_common/CMakeFiles/cpp_common.dir/get_check_data.cpp.o
FAILED: src/cpp_common/CMakeFiles/cpp_common.dir/get_check_data.cpp.o 
$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-clang++ -DBOOST_ALLOW_DEPRECATED_HEADERS -DBoost_VERSION_MACRO=108200 -DPGSQL_VERSION=161 -DPROJECT_VERSION=\"3.6.0\" -I$SRC_DIR/include -isystem $PREFIX/include/server -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/pgrouting-3.6.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -fPIC  -O3 -DNDEBUG -std=gnu++14 -isysroot /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -mmacosx-version-min=10.9   -Wsign-conversion -MD -MT src/cpp_common/CMakeFiles/cpp_common.dir/get_check_data.cpp.o -MF src/cpp_common/CMakeFiles/cpp_common.dir/get_check_data.cpp.o.d -o src/cpp_common/CMakeFiles/cpp_common.dir/get_check_data.cpp.o -c $SRC_DIR/src/cpp_common/get_check_data.cpp
$SRC_DIR/src/cpp_common/get_check_data.cpp:365:12: error: no matching function for call to 'get_array'
    return get_array(pg_array, the_size, true);
           ^~~~~~~~~
$SRC_DIR/src/cpp_common/get_check_data.cpp:265:1: note: candidate function not viable: no known conversion from 'uint64_t *' (aka 'unsigned long long *') to 'size_t *' (aka 'unsigned long *') for 2nd argument
get_array(ArrayType *v, size_t *arrlen, bool allow_empty) {
^
1 error generated.
```

--- a/include/c_types/restriction_t.h
+++ b/include/c_types/restriction_t.h
@@ -40,7 +40,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 struct Restriction_t {
     double cost;
     int64_t * via;
-    uint64_t via_size;
+    size_t via_size;
 };
 
 #endif  // INCLUDE_C_TYPES_RESTRICTION_T_H_

--- a/include/cpp_common/get_check_data.hpp
+++ b/include/cpp_common/get_check_data.hpp
@@ -59,7 +59,7 @@ char getChar(const HeapTuple, const TupleDesc&, const Column_info_t&, bool, char
 int64_t* get_array(ArrayType*, size_t*, bool);
 
 /** @brief Function returns the values of specified columns in array. */
-int64_t* getBigIntArr(const HeapTuple, const TupleDesc&, const Column_info_t&, uint64_t*);
+int64_t* getBigIntArr(const HeapTuple, const TupleDesc&, const Column_info_t&, size_t*);
 
 /** @brief Function returns the value of specified column in integer type.  */
 int64_t getBigInt(const HeapTuple, const TupleDesc&, const Column_info_t&);

--- a/src/cpp_common/get_check_data.cpp
+++ b/src/cpp_common/get_check_data.cpp
@@ -354,7 +354,7 @@ int64_t* getBigIntArr(
         const HeapTuple tuple,
         const TupleDesc &tupdesc,
         const Column_info_t &info,
-        uint64_t *the_size) {
+        size_t *the_size) {
     bool is_null = false;
 
     Datum raw_array = SPI_getbinval(tuple, tupdesc, info.colNumber, &is_null);

